### DISCUSSION TITLE
An option to automatically calibrate your last play after playing the song.

### DISCUSSION THREAD

**Discussion by TomG16** [ppy/osu#17652] [Ideas] [answered]
It occurs frequently that I want to calibrate the song it's just that I forget about it when finishing the song while it needed to be calibrated. After exiting the scores screen and not retrying it doesn't allow you to calibrate again without playing again.

Having it automatically calibrate the song for me every time I finish a song would be handy way to keep them calibrated

(Or having it remember the calibrations but I don't think that would be feasible having the calibrations stored somewhere temporarily which is bad for performance hence why auto-calibration.)

**Comment by peppy** [comment, marked as answer]
I'm not sure about making this automatic.. would giving the ability to adjust next time you start playing from your last historical play be enough to alleviate the issue? ie. remember the last play rather even when not immediately retrying?

**Reply by Sublimelime** [reply]
If it doesn't cause too much overhead/storage issues, it'd be nice to keep the "auto-offset" button available for longer yeah. A pain point I often ran into was exiting the beatmap back to song select instead of quick-retrying, and being unable to hit the auto-offset button because it had already discarded my last play's hit error.

**Reply by TomG16** [reply]
Yeah if that's feasible then I think that would fix the issue.

**Comment by respektive** [comment]
I think a button on the results screen (where it shows the UR and how much you were off) to calibrate after finishing the song would be a better way. Automatic sounds very bad... it could potentially mess up the offset more than it does good.

Remembering the last play like peppy suggested would work I think, but i think its still likely you forget about setting it when playing the song after some time again, rather than just do it on the results screen the moment you finished it.

**Reply by TomG16** [reply]
What could also work is that it saves the calibration until the next time you make a play. It then clears the calibration and replaces it with the calibration play you just made. This will make it lighter on the memory and makes it so you won't lose all your calibration data when you leave the score screen.

**Comment by AuracleTech** [comment]
I was about to open up this thread

Sometimes I forget to press retry after playing a map
Then all data is lost...

could be nice to have adjust offset on the end menu
